src/main/resources/templates/login.html

```diff
@@ -0,0 +1,20 @@
+<!DOCTYPE html>
+<html xmlns="http://www.w3.org/1999/xhtml" xmlns:th="https://www.thymeleaf.org"
+      xmlns:sec="https://www.thymeleaf.org/thymeleaf-extras-springsecurity3">
+<head>
+    <title>Spring Security Example </title>
+</head>
+<body>
+<div th:if="${param.error}">
```

Comment:
Potrafisz wyjaśnić co robią te atrybuty `th:`? NIe korzystałem z tego i nie wiem - wiesz co je wypełnia? kiedy pojawi się coś w `param.error` ?

Reply:
Szczerze to robiłem to do dwunastej w nocy na podstawie tutoriala próbowałem przerobić swój program tak by działał ale nie chciał i w ferworze walki podmieniłem poma i rowaliłem cały projekt wiec zaczołem od nowa nowe repozytorium nowy Sprinng installer i poleciałem kopjuj wklej z tutoriala. on używa biblioteki thymeleaf.org i z tego co wiem to ma ona zapewnić dynamiczną wymiane treści na stronie HTML czyli jeśli warunek jest spełniony i urzytkownik wpisał nieprawidłowe hasło wyświetli wiersz poniżej
<title>Spring Security Example </title>
</head>
<body>
<div th:if="${param.error}">
    Invalid username and password.